### PR TITLE
Project(Task) : Implement Rate Limiting for Issue Creation

### LINKED ISSUE #888: Task: implement rate limiting for posting issues and in general for the website / api

### PR DESCRIPTION
**Pull request by JisanAR03**
**What's New:**
We've added a new feature to limit how many issues users and visitors can create each day. This helps prevent spam and keeps things fair for everyone.

**Details:**

- Limits: Registered users can create up to 50 issues per day, and visitors can create up to 30.

- Tracking: We use Django's database cache to keep track of how many issues each person has made.

- Feedback: If someone tries to create too many issues, we show a friendly message about the limit.

**Why We Did This:**
To stop spam and make sure everyone gets a fair chance to report issues. We chose our own rate limiting instead of using built-in API rate limits because we needed it for both forms and APIs, not just APIs.

**How It Works:**
We added a check in the issue creation process to see how many issues someone has already created that day.
If they've reached their limit, we let them know and stop them from creating more issues until the next day.
This check is done using Django's built-in caching system, which is fast and reliable.
**Testing:**
We tested to make sure the limits work correctly for both logged-in users and visitors, and that the system resets properly every day.

**For the Team:**
After updating the site with this new feature, run `python manage.py makemigrations
python manage.py migrate` to set up everything. 

We're excited to see how this new feature helps improve our site. @DonnieBLT sir, Let me know if you have any feedback or questions!

fixed #888

### CODE REVIEW

**Review by fredfalcon — CHANGES_REQUESTED** [review]
..

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
This is a great solution thank you!  I feel we can simplify it a bit more by adding a new field on the Issue model called reporter_ip_address and then check that field for rate limiting based on the created time. This way we won't need a new table.

**Comment by JisanAR03** [comment]
> This is a great solution thank you! I feel we can simplify it a bit more by adding a new field on the Issue model called reporter_ip_address and then check that field for rate limiting based on the created time. This way we won't need a new table.

ok sir sure. I'm working on it and will come with the solution . and thanks a lot for the feedback 😊

**Comment by JisanAR03** [comment]
> This is a great solution thank you! I feel we can simplify it a bit more by adding a new field on the Issue model called reporter_ip_address and then check that field for rate limiting based on the created time. This way we won't need a new table.

@DonnieBLT sir, thank you so much for the suggestion. I updated the PR. This suggestion is very helpful. I'm just shocked. This is very simple approach and I make it very complicated 😬

storing an IP in database screenshot: 
image

**Review by fredfalcon — CHANGES_REQUESTED** [review]
two tweaks, looks good otherwise @JisanAR03

File: blt/settings.py [hunk 1]
```diff
@@ -199,6 +199,8 @@
 #        'LOCATION': 'cache_table',
 #    }
 # }
+
```
[the remark this answers is not in the document]
Comment (JisanAR03) [reply blt/settings.py:202]: updated sir

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**